backend/core/management/commands/populate_db_from_json.py

```diff
@@ -0,0 +1,385 @@
+
+        Raises
+        ------
+        CommandError
+            If the file doesn't exist, contains invalid JSON, or fails schema validation.
+        """
+        json_file = str(options["json_file"])
+        clear = bool(options.get("clear", False))
+
+        # Check if input file exists.
+        if not os.path.exists(json_file):
+            raise CommandError(f"File {json_file} does not exist")
+
+        # Load and parse JSON data.
+        try:
+            # Resolve the absolute path to handle any relative paths
+            json_path = os.path.abspath(json_file)
+            self.stdout.write(self.style.SUCCESS(f"Loading data from: {json_path}"))
+            with open(json_path, "r", encoding="utf-8") as f:
+                data = json.load(f)
+        except FileNotFoundError:
+            self.stderr.write(self.style.ERROR(f"File not found: {json_file}"))
+            self.stderr.write(
+                self.style.ERROR(f"Current working directory: {os.getcwd()}")
+            )
+            return
+        except json.JSONDecodeError as e:
+            self.stderr.write(self.style.ERROR(f"Invalid JSON in {json_file}: {e}"))
+            return
+        except Exception as e:
+            self.stderr.write(self.style.ERROR(f"Error reading file {json_file}: {e}"))
+            return
+
+        # Validate against schema.
+        try:
+            validate(instance=data, schema=SCHEMA)
+            self.stdout.write(self.style.SUCCESS("Schema validation passed"))
+        except ValidationError as e:
+            raise CommandError(f"Validation error: {e}")
+
+        self.stdout.write("Processing data...")
+
+        # MARK: - Process Data Sections.
+        if "organizations" in data:
+            self._process_organizations(data["organizations"], clear)
+
+        # Process groups.
+        if "groups" in data:
+            self._process_groups(data["groups"], clear)
+
+        # Process events.
+        if "events" in data:
+            self._process_events(data["events"], clear)
+
+        # Process global resources.
+        if "resources" in data:
+            self._process_global_resources(data["resources"], clear)
+
+        self.stdout.write(self.style.SUCCESS("Successfully populated database"))
+
+    def _process_organizations(
+        self, organizations: List[Dict[str, Any]], clear: bool
+    ) -> None:
+        """
+        Import and process organization data from the provided JSON.
+
+        Parameters
+        ----------
+        organizations : List[Dict[str, Any]]
+            List of organization data to process.
+        clear : bool
+            Flag indicating whether to clear existing data before processing.
+        """
+        self.stdout.write(f"Processing {len(organizations)} organizations...")
+        if clear:
+            self.stdout.write("Clearing existing organizations...")
+            # Add code to clear existing organizations.
```

Comment:
note: Would have been good if this code had been included in the original commits :)